src/index.ts

```diff
@@ -1,4 +1,78 @@
 import { Probot } from "probot";
+import fs from 'fs';
+import { V1Secret } from '@kubernetes/client-node/dist/gen/model/v1Secret';
+const k8s = require('@kubernetes/client-node');
+const { createAppAuth } = require("@octokit/auth-app");
+var cron = require('node-cron');
+
+const kc = new k8s.KubeConfig();
+kc.loadFromDefault();
+const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
+const k8sContext = kc.getCurrentContext();
+const k8snamespace = (() => {
+  if (k8sContext === 'inClusterContext') {
+    return fs.readFileSync('/var/run/secrets/kubernetes.io/serviceaccount/namespace', 'utf8');
+  } else {
+    return k8sContext.split('/')[0];
+  }
+})();
+console.info(`Operating in the namespace: ${k8snamespace}`);
+
+async function getToken(installId: Number) {
+
+  const appSecret = await k8sApi.readNamespacedSecret("peribolos-as-services", k8snamespace);
+  console.info(`Got the secret peribolos-as-service secrets`);
+  const appID = atob(appSecret.body.data.app_id);
+  const privateKey = atob(appSecret.body.data.private_key);
+  const auth = await createAppAuth({ appId: appID, privateKey: privateKey });
+  const appAuth = await auth({ type: "installation", installationId: installId });
+
+  return [appAuth.token, appAuth.expiresAt];
+};
+
+const patchSecret = async() => {
+
+  const resSecret = await k8sApi.listNamespacedSecret(k8snamespace,undefined,undefined,undefined,undefined,"app.kubernetes.io/created-by=peribolos")
+  for (var sec of resSecret.body.items){
+
+    console.info(`Metadata of the secret: ${sec.metadata.name}`);
+    const inId = sec.metadata.name.split("-")[1]
+    const resToken = await getToken(inId);
+
+    const current_date = new Date();
+    const expiry_date = new Date(sec.metadata.annotations.expiresAt);
+
+    // check if expiry time is less current + constant
+    if (expiry_date.getTime() < (current_date.getTime() + 40 * 60000)){
```

Comment:
40*60000 should be some kind of a constant, not a magic number.

Reply:
Not a magic number.

I thought we agreed on running the cronjob every 40 mins.
As there could be an installation in between these 40 mins time periods, I thought it would be good to check if the expiry happens in the next following 40mins, if not wait for the next cron job to update the token.

for example:
The first cronjob runs: 1:00
The next cronjob runs: 1:40
suppose at 1:35 a new install has happened, it won't make sense to update the token, as it would expire at 2:35, and the following 2:20 cronjob can update the token.

please let me know if this process is flawed.

Reply:
nono, I'm not disputing the logic here :slightly_smiling_face: I just think calculating `40*60000` is a [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) that should be named as a constant somewhere. Like

```
const EXPIRATION_THRESHOLD = 40*60000
```

or something like that

Reply:
ah i see, ack
thanks for tips